c/driver/common/utils.c

```diff
@@ -193,8 +193,10 @@ void AppendErrorDetail(struct AdbcError* error, const char* key, const uint8_t*
     details->capacity = new_capacity;
   }
 
-  char* key_data = strdup(key);
+  char* key_data = malloc(strlen(key) + 1);
   if (!key_data) return;
+  snprintf(key_data, strlen(key), "%s", key);
```

Comment:
nit: why not just `memcpy`?

Reply:
Will give that a look tomorrow. Tried strcpy originally but cpplint did not appreciate that